src/apps/fuzzers/fuzzerConstructCell.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2025 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file
+ * @brief Fuzzer program for constructCell
+ */
+
+#include "aflHarness.h"
+#include "h3api.h"
+#include "utility.h"
+
+#define STRING_LENGTH 32
+
+typedef struct {
+    int res;
+    int baseCellNumber;
+    int digits[MAX_H3_RES];
+} inputArgs;
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+    if (size < sizeof(inputArgs)) {
+        return 0;
+    }
+    const inputArgs *args = (const inputArgs *)data;
+
+    H3Index out;
+    H3_EXPORT(constructCell)
+    (args->res, args->baseCellNumber, args->digits, &out);
+    if (args->res == 0) {
+        H3_EXPORT(constructCell)
+        (args->res, args->baseCellNumber, NULL, &out);
+    }
```

Comment:
Just a (probably naive) question to improve my understanding of how to set up the fuzzers: I'm surprised we run an additional test if `res == 0`, as opposed to branching to a separate test on the value of `res`.

How does that work / why do we want to run both?

How would this compare to code like:

```c
if (args->res == 0) {
    H3_EXPORT(constructCell)
    (args->res, args->baseCellNumber, NULL, &out);
} else {
    H3_EXPORT(constructCell)
    (args->res, args->baseCellNumber, args->digits, &out);
}
```

Or, is the idea that the `res == 0` branch provides a more targeted test that's more likely to find issues quickly because it is reducing the search space?

Reply:
I think if res is 0 and a non-NULL pointer is passed for digits it should still be accepted. The inverse is not true however, since the caller must provide an array of length `res`.